portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -68,7 +59,35 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(comment);
+      int max = getRequestNum(request);  
+      response.sendRedirect("/index.html?limit=" + max);
+  }
+
+  private int getRequestNum(HttpServletRequest request) {
+      String limitString = request.getParameter("limit");
+      try {
+          int limit = Integer.parseInt(limitString);
+      } catch(NumberFormatException e) {
+          return defaultMaxComments;
+      }
+      return limit;
```

Comment:
Code does not compile: limit is not declared in this scope since it's declared in the try scope.